tools/integration_tests/util/benchmark_setup/benchmark_setup.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package benchmark_setup
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+type Benchmark interface {
+	SetupB(*testing.B)
+	TeardownB(*testing.B)
+}
+
+func getBenchmarkFunc(b *testing.B, xv reflect.Value, name string) func(*testing.B) {
+	if m := xv.MethodByName(name); m.IsValid() {
+		if f, ok := m.Interface().(func(*testing.B)); ok {
+			return f
+		}
+		// Method exists but has the wrong type signature.
+		b.Fatalf("benchmark function %v has unexpected signature (%T)", name, m.Interface())
+	}
+	return func(*testing.B) {}
+}
+
+// RunBenchmarks runs all "Benchmark*" functions that are members of x as sub-benchmarks
+// of the current benchmark. SetupB is run before the benchmark function and TeardownB is
+// run after each benchmark.
+// x must extend Benchmark interface by implementing SetupB and TearDownB methods.
+func RunBenchmarks(b *testing.B, x Benchmark) {
```

Comment:
Can we use stretchr testify to do this?

Reply:
In stretchr framework, suite.Run doesn't accept *testing.B. Although, there is a [workaround](https://github.com/stretchr/testify/issues/811#issuecomment-1328662818) but it's not clean.